subprojects/core/src/integTest/groovy/org/gradle/internal/progress/BuildProgressLoggerIntegrationTest.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.progress
+
+import org.gradle.integtests.fixtures.AbstractConsoleFunctionalSpec
+import org.gradle.test.fixtures.ConcurrentTestUtil
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.junit.Rule
+
+class BuildProgressLoggerIntegrationTest extends AbstractConsoleFunctionalSpec {
+    private static final String SERVER_RESOURCE = 'test-resource'
+
+    @Rule
+    BlockingHttpServer server = new BlockingHttpServer(40000)
```

Comment:
:o: I am not sure if the explicit timeout is needed here. I guess we'd have to try on CI.